swap_meet/vendor.py

```diff
@@ -1,2 +1,107 @@
+                return item_popped
+        return False
+
+    def get_by_id(self, id):
+        '''
+        This method finds and returns an item with a given id from the inventory list.
+        Input: id of item to be returned from the inventory list
+        Output: The item with the desired id. Returns None if the id given does not match any id of any item in the list.
+        '''
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        '''
+        Removes an item from Vendor inventory and adds it to another instance's inventory. Removes an item from other_vendor's inventory and adds it to the Vendor inventory.
+        Inputs: other_vendor (an instance of Vendor), my_item (an instance of Item), their_item (an instance of Item)
+        Outputs: Returns True if swap was successful, False if the input items are not already items in original inventories.
+        '''
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        self.add(other_vendor.remove(their_item))
+        other_vendor.add(self.remove(my_item))
+        return True
+
+    def swap_first_item(self, other_vendor):
+        '''
+        Swaps the first items of two lists of inventories.
+        Inputs: other_vendor (an instance of Vendor)
+        Output: Returns True after swapping the first items in two lists of vendor inventories. False if either itself or other_vendor has an empty inventory.
+        '''
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        self.add(other_vendor.remove(other_vendor.inventory[0]))
+        other_vendor.add(self.remove(self.inventory[0]))
+        return True
+
+    def get_by_category(self, category):
+        '''
+        Returns a list of item from inventory in a specific category
+        Input : category of the item (string)
+        Outputs: list of item in a specific category, Returns empty list if there is no item in the category.
+        '''
+        category_list = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_list.append(item)
+
+        return category_list       
+                
+    def get_best_by_category(self, category):
+        '''
+        Returns a best item in a category
+        Input : category of the item (string)
+        Outputs: return item with best category, Returns None if there is no item in the category.
+        '''
+        highest_condition = 0
+        best_category_item = None
+        for item in self.inventory:
+            if item.get_category() == category and item.condition > highest_condition:
+                highest_condition = item.condition
+                best_category_item = item
+        return best_category_item        
```

Comment:
Nice work here! Consider adding a few blank lines within your function to separate logically distinct sections of the code (e.g., setup, processing, return).  According to [PEP 8](https://peps.python.org/pep-0008/#blank-lines), blank lines can improve readability by visually grouping related operations. Right now, everything runs together, which can make it harder to quickly understand the structure and flow of your function.


```suggestion
        highest_condition = 0
        best_category_item = None
        
        for item in self.inventory:
            if item.get_category() == category and item.condition > highest_condition:
                highest_condition = item.condition
                best_category_item = item
                
        return best_category_item        
```